astroid/inference.py

```diff
@@ -837,7 +843,7 @@ def _do_compare(
     >>> _do_compare([1, 3], '<=', [2, 4])
     util.Uninferable
     """
-    retval = None
+    retval: Union[Literal[None], bool, Type[util.Uninferable]] = None
```

Comment:
```suggestion
    retval: Union[None, bool] = None
```
`Literal[None]` is not necessary. Both mypy and pyright understand `None`.

Additionally, `Uninferable` is never assigned to `retval`, only returned directly.

Reply:
Hmm there a quite a bit of `Literal[None]`'s in the codebase. Perhaps we should update those at some point.